go-controller/pkg/ovn/controller/egress_services/egress_services_controller.go

```diff
@@ -405,8 +441,20 @@ func (c *Controller) repair() error {
 		c.services[key].v6Endpoints.Insert(v6ToAdd...)
```

Comment:
I'm not sure about splitting the func as I feel it's quite handy to have the variables that are used for comparison together (i.e not needing to pass them between functions) + I think it would result in something a bit messier if we split them (will be harder to read imo). Is there something in particular you'd rather see implemented/splitted (or maybe commented more?)
I did notice thanks to your comment that the error handling isn't really good there, so I now aggregate the errors from actions on the cluster and return only in the end (there's no reason for the order you mentioned).